apps/emqx/test/emqx_session_SUITE.erl

```diff
@@ -165,15 +165,37 @@ t_publish_qos2(_) ->
 
 t_publish_qos2_with_error_return(_) ->
     ok = meck:expect(emqx_broker, publish, fun(_) -> [] end),
-    Session = session(#{
-        max_awaiting_rel => 2,
-        awaiting_rel => #{1 => ts(millisecond)}
-    }),
-    Msg = emqx_message:make(clientid, ?QOS_2, <<"t">>, <<"payload">>),
-    {error, ?RC_PACKET_IDENTIFIER_IN_USE} = emqx_session:publish(clientinfo(), 1, Msg, Session),
-    {ok, [], Session1} = emqx_session:publish(clientinfo(), 2, Msg, Session),
-    ?assertEqual(2, emqx_session:info(awaiting_rel_cnt, Session1)),
-    {error, ?RC_RECEIVE_MAXIMUM_EXCEEDED} = emqx_session:publish(clientinfo(), 3, Msg, Session1).
+    ok = meck:expect(emqx_hooks, run, fun
+        ('message.dropped', [Msg, _By, Reason]) ->
+            persistent_term:put({'message.dropped', Reason}, Msg);
+        (_Hook, _Arg) ->
+            ok
+    end),
+
+    Session = session(#{max_awaiting_rel => 2, awaiting_rel => #{PacketId1 = 1 => ts(millisecond)}}),
+    begin
```

Comment:
Q: are those two `begin/end` blocks needed here?

Reply:
for readable.
there are two steps to test `PACKET_ID_IN_USE` and `RECEIVE_MAXIMUM_EXCEEDED`